src/simple_commander/main.py

```diff
@@ -188,13 +189,15 @@ def __init__(self, x, y, angle, bonus=0, speed=0, life_count=3, type='',
         super(Hero, self).__init__(x, y, angle, bonus, speed, type, bullet_type, dimension, controller=controller)
         self.life_count = life_count
 
-    def decrease_life(self, units):
+    def decrease_life(self):
         if self.life_count > 1:
             self.life_count -= 1
+            self.response('update')
         else:
             self.life_count = 0
             self.is_dead = True
-            units.remove(self)
+            self.response('update')
+            get_game().remove_unit(self.id)
```

Comment:
by moving units removal from game field I meant to actually move logic of remove_unit() to the scope of GameController. In future I'd like to get rid of this get_game() and global game variable